internal/controller/pkg/revision/reconciler.go

```diff
@@ -474,7 +476,13 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	}
 
 	// Parse package contents.
-	pkg, err := r.parser.Parse(ctx, rc)
+	pkg, err := r.parser.Parse(ctx, struct {
```

Comment:
do you have a quick pointer to this golang syntax here? we're replacing `rc` with an anonymous struct that is composed of two anonymous fields that fit the shape of 2 interfaces....it makes sense, but just curious to read about this specific idiom/syntax and what their official nomenclature is 😂

Reply:
not sure there is a name for it, it's just a shorthand to avoid having to declare a type "LimitReaderCloser" used only here. We could argue whether it would be more idiomatic to pass a Reader in and deal with the Close part from the caller directly, but I didn't see the reason to change the code so much just for this.

Reply:
was just [looking it up](https://golangbyexample.com/embedding-interfaces-go/) now to refresh myself on the terminology - i think the right terminology here is that we're using an anonymous struct with embedded interfaces as unnamed/anonymous fields 🤓